src/mpl2/src/object.cpp

```diff
@@ -479,32 +479,50 @@ void Cluster::setParent(Cluster* parent)
   parent_ = parent;
 }
 
-void Cluster::addChild(Cluster* child)
+void Cluster::addChild(std::unique_ptr<Cluster> child)
 {
-  children_.push_back(child);
+  children_.push_back(std::move(child));
 }
 
-void Cluster::removeChild(const Cluster* child)
+std::unique_ptr<Cluster> Cluster::releaseChild(const Cluster* candidate)
```

Comment:
Is there a reason you need to use child_index and not a regular iterator from find_if?  It feels like this could be simpler.

Reply:
It could be simpler, I'm fixing it.